.github/workflows/version-compatibility.yml

```diff
@@ -0,0 +1,18 @@
+name: Validate Schema Compatibility
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  validate_schema_compatibility:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+
+      - name: Get protolock
+        run:  go install github.com/nilslice/protolock/cmd/protolock@latest
+
+      - name: Validate no breaking changes
+        run: protolock status --uptodate --protoroot ./proto/ -lockdir ./proto/
```

Comment:
Note that the `uptodate` option means that the status check will fail if there are any differences between the proto file and the proto.lock. This will help us ensure we continually keep the proto.lock up to date but does imply all contributors to the schema will need to update the proto file and then make sure the proto.lock is updated too.